portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -84,32 +85,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) {
                 return;
 
             }
-            Entity emptyAnalysisEntity = new Entity("Analysis");
-            datastore.put(emptyAnalysisEntity);
-            Entity videoEntity = Video.videoToDatastoreEntity(video, emptyAnalysisEntity.getKey());
+            Entity videoEntity = Video.videoToDatastoreEntity(video);
             datastore.put(videoEntity);
-            //Get the Datastore key of the the entity we just created as a string 
-            String videoEntityKey = KeyFactory.keyToString(videoEntity.getKey());
-            //Send key back to the client.
-            response.setContentType("text/plain");
-            try {
-                response.getWriter().println(videoEntityKey);
-            } catch (IOException e) {
-                if (DEBUG) {
-                    e.printStackTrace();
-                }
-            }
-        } else {
-            //Get entity key and send back to the client.
-            String queryVideoEntityKey = KeyFactory.keyToString(queryVideoEntity.getKey());
-            response.setContentType("text/plain");
-            try {
-                response.getWriter().println(queryVideoEntityKey);
-            } catch (IOException e) {
-                if (DEBUG) {
-                    e.printStackTrace();
-                }
-            }
+            // Create a new analysis entry using custom keys.
+            Entity videoAnalysis = new Entity("Analysis", 1L, videoEntity.getKey());
```

Comment:
When parameter values aren't clear what they are being used for, we have a convention of detailing the parameter name as a comment.

Entity videoAnalysis = 
    new Entity(/* name= */ "Analysis", /* fieldName2= */ 1L, /* fieldName3= */ videoEntity.getKey());

Reply:
Good point! Fixed.